tests/features/camel-jaxb/src/main/java/org/apache/karaf/camel/test/CamelJaxbRouteSupplier.java

```diff
@@ -0,0 +1,101 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.apache.karaf.camel.test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.model.RouteDefinition;
+import org.apache.camel.model.dataformat.JaxbDataFormat;
+import org.apache.karaf.camel.itests.AbstractCamelSingleFeatureResultMockBasedRouteSupplier;
+import org.apache.karaf.camel.itests.CamelRouteSupplier;
+import org.osgi.service.component.annotations.Component;
+
+import jakarta.xml.bind.annotation.XmlAccessType;
+import jakarta.xml.bind.annotation.XmlAccessorType;
+import jakarta.xml.bind.annotation.XmlElement;
+import jakarta.xml.bind.annotation.XmlRootElement;
+
+@Component(
+        name = "karaf-camel-jaxb-test",
+        immediate = true,
+        service = CamelRouteSupplier.class
+)
+public class CamelJaxbRouteSupplier extends AbstractCamelSingleFeatureResultMockBasedRouteSupplier {
+
+    public static final String XML_SAMPLE_NAME = "Jhon";
+    public static final int XML_SAMPLE_AGE = 33;
+
+    @Override
+    protected boolean consumerEnabled() {
+        return false;
+    }
+
+    @Override
+    public void configure(CamelContext context) {
+        JaxbDataFormat jaxb = new JaxbDataFormat();
+        jaxb.setContextPath(MyData.class.getName()); // org.apache.karaf.camel.test.CamelJaxbRouteSupplier$MyData
+        jaxb.setContextPathIsClassName("true");
+        jaxb.setPrettyPrint("false");
+        jaxb.setFragment("true"); // don't generate the XML declaration header
+
+        context.getRegistry().bind("jaxbBean", jaxb);
+    }
+
+    @Override
+    protected void configureProducer(RouteBuilder builder, RouteDefinition producerRoute) {
+        producerRoute.log("Will unmarshal: ${body}")
+            .unmarshal("jaxbBean")
+            .log("Unmarshal: ${body}")
+            .process(ex -> {
+                MyData data = ex.getIn().getBody(MyData.class);
+                assertEquals(XML_SAMPLE_NAME, data.getName());
+                assertNull(data.getNikname());
+                assertEquals(XML_SAMPLE_AGE, data.getAge());
+            }).log("Will marshal: ${body}")
+            .marshal("jaxbBean")
+            .log("Marshal: ${body}")
+            .toF("mock:%s", getResultMockName());
```

Comment:
I don't see the reason why I added these extra tests here: https://github.com/apache/camel-karaf/blob/4194d52519783707e16faa88916572549907e4df/tests/features/camel-jackson/src/main/java/org/apache/karaf/camel/test/CamelJacksonRouteSupplier.java#L97
... since the exect same input is being tested here: https://github.com/apache/camel-karaf/blob/4194d52519783707e16faa88916572549907e4df/tests/features/camel-jackson/src/test/java/org/apache/karaf/camel/itest/CamelJacksonITest.java#L41

Should I rectify my `jaxson` and `jacksonxml` tests in this PR?

Reply:
Agree that you could keep only the `mock.expectedBodiesReceived(JSON_SAMPLE);`